ibis/expr/datatypes/cast.py

```diff
@@ -46,166 +43,3 @@ def highest_precedence(dtypes: Iterator[dt.DataType]) -> dt.DataType:
         return functools.reduce(higher_precedence, collected)
     else:
         return dt.null
-
-
-@castable.register(dt.DataType, dt.DataType)
```

Comment:
Shall we have function for backward compatibility?

Reply:
I don't think so, this isn't a public API.

Reply:
How about removing `dt.cast()` as well in favor of the `cast()` method?